mlrun/api/crud/secrets.py

```diff
@@ -11,40 +13,59 @@
 
 class Secrets(metaclass=mlrun.utils.singleton.Singleton,):
     internal_secrets_key_prefix = "mlrun."
+    # make it a subset of internal since key map are by definition internal
+    key_map_secrets_key_prefix = f"{internal_secrets_key_prefix}map."
 
     def generate_schedule_secret_key(self, schedule_name: str):
         return f"{self.internal_secrets_key_prefix}schedules.{schedule_name}"
 
+    def generate_schedule_key_map_secret_key(self):
+        return f"{self.key_map_secrets_key_prefix}schedules"
+
+    @staticmethod
+    def validate_secret_key_regex(key: str, raise_on_failure: bool = True) -> bool:
+        return mlrun.utils.helpers.verify_field_regex(
+            "secret.key", key, mlrun.utils.regex.secret_key, raise_on_failure
+        )
+
+    def validate_internal_secret_key_allowed(
+        self, key: str, allow_internal_secrets: bool = False
+    ):
+        if self._is_internal_secret_key(key) and not allow_internal_secrets:
+            raise mlrun.errors.MLRunAccessDeniedError(
+                f"Not allowed to create/update internal secrets (key starts with "
+                f"{self.internal_secrets_key_prefix})"
+            )
+
     def store_secrets(
         self,
         project: str,
         secrets: mlrun.api.schemas.SecretsData,
         allow_internal_secrets: bool = False,
+        key_map_secret_key: typing.Optional[str] = None,
+        allow_key_maps: bool = False,
```

Comment:
This got me a bit confused, since I assumed it means whether you allow using key-maps when storing keys in general. Maybe rename it to `allow_storing_key_maps`